src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -154,7 +154,7 @@ public EC2AbstractSlave(String name, String instanceId, String description, Stri
         this.stopOnTerminate = stopOnTerminate;
         this.idleTerminationMinutes = idleTerminationMinutes;
         this.tags = tags;
-        this.usePrivateDnsName = connectionStrategy.equals(ConnectionStrategy.PRIVATE_DNS);
+        this.usePrivateDnsName = connectionStrategy != null && connectionStrategy.equals(ConnectionStrategy.PRIVATE_DNS);
```

Comment:
If this is an enum, you can just do:

```suggestion
        this.usePrivateDnsName = connectionStrategy == ConnectionStrategy.PRIVATE_DNS;
```